titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/common/action/task/KillInitiatedActions.java

```diff
@@ -203,7 +202,20 @@ public static ChangeAction reconcilerInitiatedTaskKillInitiated(ReconciliationEn
                         JobManagerConstants.RECONCILER_CALLMETADATA.toBuilder().withCallReason(reason).build()
                 ));
             }
-        });
+        }
+
+        // Move running tasks to KillInitiated state
+        for (EntityHolder taskHolder : runningView.getChildren()) {
+            if(result.size() >= concurrencyLimit) {
```

Comment:
test to validate concurrency limit is honored?

Reply:
Added